### PR TITLE
Added scrolling and massive refactoring

### LINKED ISSUE #248: Properly Load and Display old Messages
**Is your feature request related to a problem? Please describe.**
User should see old messages when he clicks on a chat

**Describe the solution you'd like**
Somehow load some old messages in the background, so that when a chat is clicked, the first ca 30 messages are already loaded and user doesnt have to wait.

Also make the old message list scrollable.

Maybe We should load first 30messages of a chat directly when a user loggs in for the first time and then these messages are locally cached in the tdlib and can quickly be displayed the next time?

### PR DESCRIPTION
**Pull request by schmidtjonas**
closes #248

### CODE REVIEW

File: packages/TelegramClient-Core.package/TCCChat.class/instance/initialize.st [hunk 1]
```diff
@@ -1,4 +1,7 @@
 accessing
 initialize
 
-	self messages: OrderedCollection new.
+	self
+		messages: (OrderedCollection newFrom: {TCCNullMessage new});
+		waitingForUpdate: false;
+		numberOfRequestedMessages: 20.
```
Comment (rohansaw) [inline packages/TelegramClient-Core.package/TCCChat.class/instance/initialize.st:7]: This sould be in extra method no magic numbers please

File: packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/chatHistoryReceived..st [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
 
-		chat messages ifNotEmpty: [
-			oldestMessageId := chat messages last messageId.
-			chatMessageNumber := self messageLimit - chat messages size.
-			chatMessageNumber > 0 ifTrue: [self getChatHistoryFrom: chatID with: oldestMessageId and: chatMessageNumber]]].
\ No newline at end of file
+		chat waitingForUpdate: false.
+		
+		chat stillRequestedMessages > 0 
+			ifTrue: [self
```
Comment (rohansaw) [inline packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/chatHistoryReceived..st:15]: spacing looks wrong

File: packages/TelegramClient-UI.package/TCUChatWindow.class/instance/showChatMessagesFrom..st [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	| messageListReversed |
+	messageListReversed := self selectedChat messages reversed.
+	messageListReversed removeLast.
+	self messageListWindow clearMessages.
+	self messageListWindow lowestVisibleMessage: lowestMessage.
+
+	1 to: messageListReversed size - lowestMessage do: [: x |
+		self messageListWindow addMessageAtBottom: (messageListReversed at: x)].
+
+	messageListReversed size < (lowestMessage + 20) ifTrue: [
```
Comment (rohansaw) [inline packages/TelegramClient-UI.package/TCUChatWindow.class/instance/showChatMessagesFrom..st:13]: Magic Number again

File: packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/chatHistoryReceived..st (hunk 2 above)
Comment (schmidtjonas) [reply packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/chatHistoryReceived..st:15]: self in new line?

Comment (rohansaw) [reply packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/chatHistoryReceived..st:15]: not sure bout that but the stuff after self should be tabbed in

File: packages/TelegramClient-UI.package/TCUChatWindow.class/instance/showChatMessagesFrom..st (hunk 3 above)
Comment (schmidtjonas) [reply packages/TelegramClient-UI.package/TCUChatWindow.class/instance/showChatMessagesFrom..st:13]: will be addressed in #260

**Pull request merged**